### PR TITLE
feat: add StorefrontLive — agent-buildable customer-facing pages

### LINKED ISSUE #76: feat: add StorefrontLive — agent-buildable customer-facing pages served by Phoenix
## Context

The admin panel is agent-native: describe what you want → UI spec → HEEx components → it appears. The hand-coded vanilla HTML/JS in `jarga-commerce/public/` is the **marketing and sales site** for the Jarga Commerce platform (jargacommerce.com) and must not be modified. This issue adds a separate **demo storefront** — a working ecommerce store served at `demo.jargacommerce.com` — built with the same generative UI spec → HEEx pipeline as the admin.

The `jarga-commerce` backend already has a complete Frontend API (`frontend-api.yaml`):
- `GET /v1/frontend/pages/{slug}` — retrieve published page content
- `POST /v1/frontend/pages` — create page draft
- `PATCH /v1/frontend/page-drafts/{page_id}` — update draft
- `POST /v1/frontend/page-drafts/{page_id}/publish` — publish
- `GET /v1/frontend/navigation` — site navigation
- `PUT /v1/frontend/navigation` — update navigation
- `GET /v1/frontend/slots/{slot_key}` — named content slots
- `POST /v1/frontend/bootstrap` — seed entire site in one call
- `POST /v1/frontend/preview/{slug}` — preview tokens

The existing admin already has the `UiSpec → Renderer → HEEx` pipeline. We extend this same pattern with **storefront component types** and a **public LiveView** that serves customer-facing pages.

---

## Design Reference: Zara Home (zarahome.com/gb)

The first storefront implementation should mirror the visual language and component patterns of **Zara Home** — an editorial, luxury-minimal ecommerce experience. This is the target aesthetic, not a pixel-perfect clone. We take the patterns that work and adapt them to Jarga's component-based architecture.

### Why Zara Home

Zara Home sits at the intersection of editorial fashion magazine and functional ecommerce. It treats product pages as curated visual stories rather than utilitarian grids. This aligns perfectly with Jarga's positioning: a premium, agent-built storefront that feels like a lifestyle brand, not a template.

Key characteristics:
- **Editorial over utilitarian** — the site feels like a curated magazine, not a product catalog
- **Photography-first** — imagery dominates, text is minimal and restrained
- **Luxury minimalism** — extreme whitespace, thin typography, monochromatic palette
- **Room-based navigation** — products organised by living context (Bedroom, Kitchen & Dining, Bathroom, Home Decor, Fragrances) rather than just product type
- **Mixed-scale layouts** — product grids use varying image sizes (large hero shots mixed with smaller tiles) rather than uniform grids
- **Quiet UI chrome** — navigation, buttons, and interactive elements are deliberately understated

### Visual Design Tokens (Zara Home Style)

| Token | Value | Notes |
|---|---|---|
| **Font: primary** | `"Helvetica Neue", Helvetica, Arial, sans-serif` | Ultra-clean sans-serif throughout |
| **Font: weight headings** | `300` (Light) | Thin, elegant headings — NOT bold |
| **Font: weight body** | `400` (Regular) | |
| **Font: weight nav** | `400` | All-caps, wide letter-spacing |
| **Letter-spacing: nav/headings** | `0.2em – 0.3em` | Wide-tracked uppercase creates the luxury feel |
| **Letter-spacing: body** | `0.02em – 0.04em` | Subtle tracking on body text |
| **Text transform: nav** | `uppercase` | All navigation and category labels |
| **Background** | `#ffffff` | Pure white — no warm tints |
| **Text: primary** | `#000000` or `#1a1a1a` | Near-black |
| **Text: secondary** | `#666666` | Product prices, secondary info |
| **Text: muted** | `#999999` | Tertiary info, timestamps |
| **Border/divider** | `#e0e0e0` or `rgba(0,0,0,0.08)` | Barely-there dividers |
| **Accent/CTA** | `#000000` | Black buttons, black text links — no colour accent |
| **Button style** | Black bg + white text, OR white bg + black border | Two button tiers |
| **Card background** | `#ffffff` | No card backgrounds — products float on white |
| **Hover state** | Underline or subtle opacity change | No colour shifts, no shadows |
| **Image treatment** | No borders, no shadows, no rounded corners | Images bleed into whitespace |
| **Max content width** | `1400px` – `1600px` | Wide layouts to showcase photography |
| **Nav bar** | White bg, black text, uppercase, fixed on scroll | Minimal, disappears on scroll-down, reappears on scroll-up |
| **Border radius** | `0` | Sharp corners everywhere — no rounding |

#### Homepage

The Zara Home homepage is a vertical scroll of full-width and split editorial blocks:

1. **Full-bleed hero** — a single atmospheric photograph (lifestyle shot of a styled room) with minimal text overlay: collection name in thin uppercase, optional "SHOP NOW" link. No carousel, no slider. One strong image.
2. **Category editorial blocks** — alternating full-width and split (50/50) image blocks, each linking to a category. "BEDROOM" over a styled bedroom shot, "KITCHEN & DINING" over a table setting. Text is always small, uppercase, tracked, overlaid on the image or placed below.
3. **New arrivals strip** — a horizontal scrolling row of product cards (image + name + price) with no borders, no backgrounds.
4. **Editorial campaign block** — a full-width cinematic image promoting a seasonal campaign or designer collaboration, with a single text link.
5. **Footer** — minimal, black or dark grey background, white text, links organised in columns (categories, company info, help, legal).

No product grids on the homepage. It's purely editorial — curated imagery that drives you into categories.

#### Collection/Category Page

1. **Collection header** — full-width lifestyle photograph with category name overlaid in thin uppercase
2. **Subcategory navigation** — horizontal text links (e.g., "DUVET COVERS · PILLOWCASES · FITTED SHEETS · QUILTS") in small uppercase
3. **Product grid** — mixed-scale layout:
   - Default: 2-column or 3-column grid on desktop
   - Option to switch to a tighter 4-column grid (layout toggle icons, top-right)
   - Some images are double-height or double-width for editorial emphasis
   - Product info below each image: product name (sentence case, light weight), price
   - No borders, no cards, no shadows — products on pure white
   - On hover: second product image swaps in (lifestyle → detail, or angle change)
4. **Filter** — minimal slide-out panel from left, not a sidebar. Options: colour, size, price range. Styled as small uppercase text, not dropdowns.

#### Product Detail Page

1. **Image gallery** — dominates 60-70% of the viewport width. Vertical scroll of full-size photographs (not a carousel with dots). Mix of lifestyle/styled shots and clean product-only shots. On click: full-screen zoom.
2. **Product info panel** — right side, sticky on scroll:
   - Product name: sentence case, `font-weight: 300`
   - Price: same weight, slightly larger
   - Colour swatches: small circles, black border on selected
   - Size selector: horizontal row of size labels, outlined box on selected
   - "ADD TO BASKET" button: full-width, black bg, white text, uppercase, tracked
   - Accordion sections below: "DESCRIPTION", "COMPOSITION & CARE", "DELIVERY & RETURNS" — each expands on click, thin border-bottom dividers
3. **"WEAR WITH" / "YOU MAY ALSO LIKE"** — horizontal scroll of related products, same minimal card style

#### Cart / Basket

1. **Slide-out drawer** from right (not a separate page)
2. **Line items** — product image (small square), name, size, colour, quantity selector, price
3. **Totals** at bottom: subtotal, shipping estimate, total
4. **"CONTINUE SHOPPING"** text link + **"CHECKOUT"** black button
5. **Cross-sell suggestions** below the totals

### Component Mapping (Zara Home → Jarga Storefront)

| Zara Home Pattern | Jarga Component | Notes |
|---|---|---|
| Full-bleed hero photograph | `editorial_hero` | Full-viewport-width image, thin text overlay, optional CTA link |
| Category editorial block (50/50 split) | `editorial_split` | Two images side-by-side, each linking to a category, text overlay |
| Category editorial block (full-width) | `editorial_full` | Single atmospheric image linking to a category |
| Horizontal product scroll | `product_scroll` | Horizontally scrolling row of product cards, no wrapping |
| Mixed-scale product grid | `product_grid` | 2/3/4 column grid with optional featured (double-size) items |
| Product card (minimal) | `product_card` | Image + name + price, no border/shadow, hover image swap |
| Product detail page | `product_detail` | Image gallery (vertical scroll) + sticky info panel |
| Subcategory nav strip | `category_nav` | Horizontal text links for subcategories within a collection |
| Filter panel | `filter_drawer` | Slide-out filter panel from left edge |
| Accordion sections | `accordion` | Expandable text sections (description, care, delivery) |
| Cart drawer | `cart_drawer` | Slide-out from right, line items, totals, checkout CTA |
| Announcement bar | `announcement_bar` | Top-of-page thin banner for promotions/shipping info |
| Site navigation | `nav_bar` | Fixed top bar: logo centre, hamburger left, search + basket right |
| Footer | `footer` | Dark background, column layout, category/info/legal links |
| Newsletter signup | `newsletter_signup` | Email input + submit, minimal styling |
| "You may also like" | `related_products` | Horizontal scroll of product cards |
| Colour swatches | `colour_picker` | Small colour circles with selection indicator |
| Size selector | `size_selector` | Horizontal row of size labels |
| Image gallery (PDP) | `image_gallery` | Vertical scroll of full-width product photographs |
| Search overlay | `search_overlay` | Full-screen search with recent/suggested results |
| Breadcrumbs | `breadcrumb_nav` | Minimal text breadcrumb trail |

### Navigation Structure

Zara Home uses a **mega-menu** that expands from the hamburger or category hover:

```
SALE | BEST SELLERS | BEDROOM | HOME DECOR | KITCHEN & DINING | BATHROOM | FRAGRANCES | KIDS & BABY | FURNITURE | CLOTHING & SHOES | LIFESTYLE
```

Each top-level category expands to show subcategories in columns. The navigation should be data-driven from the Frontend API navigation endpoint, so the agent can restructure it conversationally.

### Interaction Patterns

- **Scroll-triggered nav** — nav bar visible on scroll-up, hidden on scroll-down (maximises image viewport)
- **Image hover swap** — product cards swap to a second image on hover (e.g., styled → flat-lay)
- **Smooth scroll** — page scrolls smoothly between editorial sections
- **Lazy image loading** — images load as they enter viewport (critical for image-heavy pages)
- **Minimal transitions** — no flashy animations. Subtle opacity fades (200-300ms) only.
- **Full-screen image zoom** — clicking a product image opens a full-screen overlay with zoom/pan

### Mobile Adaptations

- Product grid collapses to 2 columns (compact tiles, 4 products visible per screen)
- Navigation becomes a full-screen overlay menu
- Product detail: image gallery becomes a horizontal swipe carousel
- Cart drawer becomes a full-screen overlay
- Sticky "ADD TO BASKET" bar at bottom of product detail page

---

### Same pattern, different components

The admin uses `UiSpec.parse/1 → Renderer.render_spec/1 → JargaComponents.*`. The storefront uses:

- `StorefrontRenderer` — maps storefront specs → HEEx assigns (parallel to `Renderer`)
- `StorefrontComponents` — customer-facing HEEx components (parallel to `JargaComponents`)
- `StorefrontLive` — public LiveView that loads page spec from Frontend API and renders it

### Page content_json as UI spec

Today `content_json` in the Frontend API stores arbitrary JSON for `frontend-content.js` to hydrate. Going forward, it stores a **UI spec** with storefront component types:

```json
{
  "layout": "storefront",
  "components": [
    {
      "type": "editorial_hero",
      "data": {
        "image_url": "/media/winter-bedroom.jpg",
        "title": "WINTER COLLECTION",
        "subtitle": "Warmth meets elegance",
        "cta": {"label": "SHOP NOW", "href": "/bedroom"}
      }
    },
    {
      "type": "editorial_split",
      "data": {
        "left": {
          "image_url": "/media/kitchen-styled.jpg",
          "label": "KITCHEN & DINING",
          "href": "/kitchen-dining"
        },
        "right": {
          "image_url": "/media/bathroom-styled.jpg",
          "label": "BATHROOM",
          "href": "/bathroom"
        }
      }
    },
    {
      "type": "product_scroll",
      "data": {
        "title": "NEW ARRIVALS",
        "source": "newest",
        "limit": 12
      }
    },
    {
      "type": "editorial_full",
      "data": {
        "image_url": "/media/fragrances-campaign.jpg",
        "label": "FRAGRANCES",
        "href": "/fragrances"
      }
    }
  ]
}
```

### Dynamic data hydration

Some components reference live data (e.g. `product_grid` with `source: "featured"`). `StorefrontLive` hydrates these at render time by calling the PIM/Inventory APIs, so product data is always current. The spec defines **what** to show, the LiveView fetches **current data**.

### Router structure

```elixir
# Admin routes (existing, authenticated)
scope "/admin" do
  pipe_through [:browser, :require_auth]
  live "/chat", ChatLive
end

# Storefront routes (public, no auth)
scope "/" do
  pipe_through [:browser, :storefront]
  live "/*slug", StorefrontLive
end
```

### Agent storefront building flow

In the admin ChatLive:
1. Merchant: "Build me a homepage in the Zara Home style with a full-bleed hero and category blocks"
2. Agent calls PIM API for collections/products, generates an editorial storefront UI spec
3. Spec renders in a **preview pane** (right side of chat split, same as admin views)
4. Merchant tweaks: "Make the hero image the bedroom collection", "Add a fragrances campaign block"
5. Agent updates the spec, preview updates live
6. Merchant: "Publish it" → agent calls `POST /v1/frontend/pages` + `publish`
7. Page is live at the public route

### Editorial / Layout Components

| Component | Purpose |
|---|---|
| `editorial_hero` | Full-viewport-width atmospheric photograph with thin uppercase text overlay and optional CTA link |
| `editorial_full` | Full-width category/campaign image block with overlaid label linking to a section |
| `editorial_split` | Side-by-side (50/50) image blocks, each linking to a category with text overlay |
| `announcement_bar` | Thin top-of-page banner for shipping info or promotions |
| `nav_bar` | Fixed top bar — logo centred, hamburger left, search + basket right. Hides on scroll-down, shows on scroll-up |
| `category_nav` | Horizontal subcategory text links within a collection page (e.g., "DUVET COVERS · PILLOWCASES · SHEETS") |
| `footer` | Dark background, columns of links (categories, company, help, legal), newsletter signup, social icons |
| `breadcrumb_nav` | Minimal text breadcrumb trail |
| `text_block` | Rich text content section for about/editorial pages |

### Product Components

| Component | Purpose |
|---|---|
| `product_grid` | Mixed-scale grid (2/3/4 columns) with optional featured (double-size) items. Layout toggle. Zara Home's signature varied-size grid |
| `product_card` | Minimal card: image + name (sentence case, light weight) + price. No borders, no shadows. Hover swaps to second image |
| `product_scroll` | Horizontally scrolling row of product cards for "New Arrivals", "Best Sellers" sections |
| `product_detail` | Full product page: vertical-scroll image gallery (60-70% width) + sticky info panel (name, price, colour swatches, size selector, ADD TO BASKET, accordion sections) |
| `image_gallery` | Vertical scroll of full-width product photographs with click-to-zoom overlay |
| `colour_picker` | Small colour circles with border indicator for selected colour |
| `size_selector` | Horizontal row of size labels with outlined box for selected size |
| `related_products` | "You may also like" / "Wear with" horizontal scroll of product cards |

### Commerce Components

| Component | Purpose |
|---|---|
| `cart_drawer` | Slide-out drawer from right: line items (image, name, size, colour, qty, price), totals, checkout CTA, cross-sell suggestions |
| `search_overlay` | Full-screen search overlay with search input, recent searches, suggested products |
| `filter_drawer` | Slide-out filter panel from left: colour, size, price range, material. Minimal text styling |
| `newsletter_signup` | Email capture: input + submit button, minimal styling |

### Component design system

All components use CSS custom properties for theming (#77). **Default theme matches the Zara Home aesthetic:**

```css
:root {
  /* Typography */
  --font-primary: "Helvetica Neue", Helvetica, Arial, sans-serif;
  --font-weight-light: 300;
  --font-weight-regular: 400;
  --font-weight-medium: 500;
  --letter-spacing-nav: 0.25em;
  --letter-spacing-heading: 0.2em;
  --letter-spacing-body: 0.03em;

  /* Colours */
  --color-bg: #ffffff;
  --color-text: #1a1a1a;
  --color-text-secondary: #666666;
  --color-text-muted: #999999;
  --color-border: rgba(0, 0, 0, 0.08);
  --color-accent: #000000;
  --color-btn-primary-bg: #000000;
  --color-btn-primary-text: #ffffff;
  --color-nav-bg: #ffffff;
  --color-footer-bg: #1a1a1a;
  --color-footer-text: #ffffff;

  /* Layout */
  --max-width: 1440px;
  --border-radius: 0;
  --transition-speed: 200ms;

  /* Spacing */
  --space-xs: 8px;
  --space-sm: 16px;
  --space-md: 32px;
  --space-lg: 64px;
  --space-xl: 96px;
}
```

### Cart & checkout integration

- `cart_drawer` and `product_card` use LiveView events (`phx-click="add_to_cart"`)
- `StorefrontLive` manages a basket via the Basket API (`POST /v1/basket/items`)
- Checkout redirects to Stripe Hosted Checkout via `POST /v1/checkout/sessions`
- Basket ID stored in browser session (Phoenix session or cookie)

### SEO

- LiveView is server-rendered — full HTML on first load (good for crawlers)
- `StorefrontLive` sets `<title>`, `<meta description>`, Open Graph tags from page spec
- Structured data (JSON-LD) generated from product data for rich snippets

## Acceptance Criteria

- [ ] `StorefrontLive` LiveView that loads page spec from `GET /v1/frontend/pages/{slug}` and renders it
- [ ] `StorefrontRenderer` module that maps storefront specs → component assigns
- [ ] `StorefrontComponents` module with Zara Home-inspired components: `editorial_hero`, `editorial_full`, `editorial_split`, `product_grid`, `product_card`, `product_scroll`, `product_detail`, `nav_bar`, `footer`, `cart_drawer`, `search_overlay`, `filter_drawer`, `image_gallery`, `announcement_bar`, `category_nav`
- [ ] Default design tokens match the Zara Home aesthetic: Helvetica Neue, `font-weight: 300` headings, uppercase tracked navigation, pure white backgrounds, black accents, no border-radius, no shadows
- [ ] Editorial homepage layout: full-bleed hero → category editorial blocks → product scroll → campaign block
- [ ] Product grid with mixed-scale layout (featured/double-size items) and 2/3/4 column toggle
- [ ] Product cards with hover image swap
- [ ] Product detail page with vertical-scroll image gallery + sticky info panel + accordion sections
- [ ] Cart drawer slide-out with line items and totals
- [ ] Navigation: fixed bar with scroll-hide behaviour, mega-menu for categories
- [ ] Public routes: `/*slug` catches all storefront pages, `/` maps to slug `home`
- [ ] Dynamic data hydration: `product_grid` and `product_scroll` fetch live product data from PIM API
- [ ] Cart functionality: add-to-cart, cart drawer, basket API integration
- [ ] SEO: page title, meta description, OG tags set from page spec
- [ ] Preview mode: `StorefrontPreviewLive` accessible from admin with preview token
- [ ] Components styled with CSS custom properties (themeable via #77)
- [ ] Mobile responsive: 2-column grid, full-screen nav overlay, swipe image gallery, sticky add-to-basket bar
- [ ] `mix test` passes

## Files to create/change

- `lib/jarga_admin_web/live/storefront_live.ex` — new
- `lib/jarga_admin_web/live/storefront_preview_live.ex` — new
- `lib/jarga_admin_web/components/storefront_components.ex` — new
- `lib/jarga_admin/storefront_renderer.ex` — new
- `lib/jarga_admin_web/router.ex` — add public storefront routes
- `lib/jarga_admin/api.ex` — add basket/checkout convenience wrappers if missing
- `assets/css/storefront.css` — Zara Home-inspired styles with CSS custom properties
- `assets/js/storefront_hooks.js` — LiveView hooks for cart drawer, image gallery zoom, scroll-hide nav, image hover swap, lazy loading
- Tests for renderer, components, and StorefrontLive

## Dependencies

- Frontend API endpoints implemented in `jarga-commerce` (✅ already exist)
- Basket API endpoints (✅ already exist)
- Checkout API endpoints (✅ already exist)
- Theme system (#77) — can be built in parallel, storefront uses Zara Home defaults until theme slots are wired

## Notes

- **This is the demo store, NOT a replacement for the marketing site.** The hand-coded HTML/JS/CSS in `jarga-commerce/public/` is the marketing and sales website for the Jarga Commerce platform (jargacommerce.com). It must not be removed or modified. This storefront is a completely separate application served from `jarga-admin` at `demo.jargacommerce.com`.
- This does NOT replace the admin UI — admin routes (`/admin/*`) and storefront routes (`/*slug`) coexist in the same Phoenix app
- The same agent that manages the admin can build demo storefront pages via the chat interface
- The Zara Home aesthetic is the **default theme** — merchants can override every token via the theme system (#77) to create completely different looks
- The demo store showcases what a merchant's storefront looks like when built and managed by the Jarga agent — it is the primary sales tool for the platform
- Future: agent can autonomously build and publish storefront pages in fully autonomous mode
- Future: additional template aesthetics beyond Zara Home (e.g., "Warm Artisan", "Bold Modern", "Scandinavian Minimal") delivered as preset theme token sets

## Deployment

The demo storefront is served from the `jarga-admin` Phoenix app at **`demo.jargacommerce.com`**:

- In production: hostname-based routing — requests to `demo.jargacommerce.com` hit `StorefrontLive`, requests to `admin.jargacommerce.com` hit `ChatLive`
- In development: path-based routing — `localhost:4000/admin/*` for admin, `localhost:4000/*` for storefront
- The demo store is seeded with sample products, collections, and editorial pages that showcase the agent-built storefront experience
- The `jarga-commerce` platform marketing site (jargacommerce.com) remains a separate static site served by the Rust binary — completely independent, not touched by this work

### PR DESCRIPTION
**Pull request by quigles1977**
## Summary

Closes #76.

Adds the **storefront system** for rendering customer-facing pages at `demo.jargacommerce.com`, using the same UI spec → renderer → HEEx pipeline as the admin panel.

### Domain layer
- **`StorefrontRenderer`** — maps storefront UI specs (from `content_json`) to component assigns. Handles: `editorial_hero`, `editorial_full`, `editorial_split`, `product_scroll`, `product_grid`, `product_detail`, `announcement_bar`, `nav_bar`, `footer`, `category_nav`, `text_block`
- **API wrappers** — Frontend API endpoints (`pages`, `navigation`, `slots`, `bootstrap`, `preview`) + basket/checkout endpoints added to `Api` module

### Web layer
- **`StorefrontLive`** — public LiveView that loads page spec from `GET /v1/frontend/pages/{slug}` and renders it via component dispatch
- **`StorefrontComponents`** — 15 Zara Home-inspired HEEx components with luxury-minimal aesthetic
- **Storefront layout** — bare wrapper (no admin nav/footer chrome)
- **Router** — `/store/*` routes for dev, hostname-based routing in prod (`demo.jargacommerce.com`)

### Design system (Zara Home aesthetic)
- **CSS** — 1028 lines of Zara Home design tokens as CSS custom properties: Helvetica Neue, `font-weight: 300` headings, uppercase tracked nav, pure white backgrounds, black accents, zero border-radius
- **JS hooks** — `StorefrontNav` (scroll-hide nav), `ImageHoverSwap` (preload hover images)
- **Mobile responsive** — 2-column grid, collapsible nav, sticky add-to-basket, full-width images

### Components implemented
| Category | Components |
|----------|-----------|
| Editorial | `editorial_hero`, `editorial_full`, `editorial_split` |
| Product | `product_scroll`, `product_grid`, `product_card`, `product_detail` |
| Navigation | `announcement_bar`, `nav_bar`, `category_nav` |
| Commerce | `cart_drawer` |
| Content | `text_block`, `footer` |

## Tests
- **23 new tests** (16 unit + 7 integration), all passing
- No regressions — 261 tests total, same 20 pre-existing failures
- Renderer tests cover all component types + edge cases
- LiveView tests use Bypass to stub Frontend API responses

## Placeholder / Future work
- Cart drawer UI exists but basket API wiring is placeholder
- Search overlay is placeholder
- Hostname-based routing plug for production (`demo.jargacommerce.com`)
- Theme system (#77) will make design tokens overridable
- Multi-storefront support (#78) via Channels API

### CODE REVIEW

File: lib/jarga_admin_web/live/storefront_live.ex [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+  @impl true
+  def mount(params, _session, socket) do
+    slug = resolve_slug(params)
+
+    socket =
+      socket
+      |> assign(:page_title, "Loading…")
+      |> assign(:slug, slug)
+      |> assign(:page, nil)
+      |> assign(:components, [])
```
Comment (quigles1977) [inline lib/jarga_admin_web/live/storefront_live.ex]: [unbounded] **`@components` assign stores full component tree as a plain list — duplicates `@page` data.**

The `@components` assign holds the entire parsed component list (which includes all product data, image URLs, etc.) as a regular list in process memory. Combined with `@page` (the raw API response), the full page content is stored twice in the LiveView process. At minimum, drop `@page` as noted in the other comment to avoid storing two copies of the same content.

File: lib/jarga_admin_web/live/storefront_live.ex [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+  end
+
+  defp render_component(%{component: %{type: :unknown}} = assigns) do
+    ~H"""
+    """
+  end
+
+  # ── Data loading ──────────────────────────────────────────────────────────
+
+  defp load_page_data(socket, slug) do
```
Comment (quigles1977) [inline lib/jarga_admin_web/live/storefront_live.ex]: [hot-path] **Sequential API calls — page + navigation are independent but serialised.**

Inside `load_page_data/2`, `load_navigation()` blocks first, then `Api.get_storefront_page(slug)` is called. These two HTTP requests are independent and could be parallelised using `Task.async`/`Task.await` to cut the mount latency roughly in half:

```elixir
defp load_page_data(socket, slug) do
  nav_task = Task.async(fn -> load_navigation() end)
  page_result = Api.get_storefront_page(slug)
  nav_links = Task.await(nav_task, 5_000)
  # ... same as before
end
```

This matters because it's on the critical path — customers see a loading spinner until both calls complete.

File: lib/jarga_admin_web/live/storefront_live.ex [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+      |> assign(:slug, slug)
+      |> assign(:page, nil)
+      |> assign(:components, [])
+      |> assign(:nav_links, [])
+      |> assign(:error, nil)
+      |> assign(:cart_open, false)
+      |> assign(:cart_items, [])
+      |> assign(:cart_count, 0)
+      |> assign(:mobile_menu_open, false)
+      |> load_page_data(slug)
```
Comment (quigles1977) [inline lib/jarga_admin_web/live/storefront_live.ex]: [arch] **Synchronous API calls in mount/3 block the initial render.** `load_page_data/2` calls `Api.get_storefront_page/1` and `Api.get_storefront_navigation/0` synchronously in mount, which means the LiveView cannot render until both HTTP requests complete. If the upstream API is slow or down, the user sees a blank page (or a timeout).

The admin ChatLive avoids this by checking `connected?(socket)` and deferring data loading. Do the same here:

```elixir
def mount(params, _session, socket) do
  slug = resolve_slug(params)
  socket =
    socket
    |> assign(:page_title, "Loading...")
    |> assign(:slug, slug)
    |> assign(:loading, true)
    |> assign_defaults()

  if connected?(socket), do: send(self(), {:load_page, slug})

  {:ok, socket, layout: {JargaAdminWeb.Layouts, :storefront}}
end
```

Then handle data loading in `handle_info/2`. This gives an instant static render on first load and avoids blocking the BEAM process.
